apis/replication.storage/v1alpha1/volumegroupreplicationcontent_types.go

```diff
@@ -0,0 +1,107 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VolumeGroupReplicationContentSpec defines the desired state of VolumeGroupReplicationContent
+type VolumeGroupReplicationContentSpec struct {
+	// VolumeGroupreplicationRef specifies the VolumeGroupReplication object to which this
+	// VolumeGroupReplicationContent object is bound.
+	// VolumeGroupReplication.Spec.VolumeGroupReplicationContentName field must reference to
+	// this VolumeGroupReplicationContent's name for the bidirectional binding to be valid.
+	// For a pre-existing VolumeGroupReplicationContent object, name and namespace of the
+	// VolumeGroupReplication object MUST be provided for binding to happen.
+	// This field is immutable after creation.
+	// Required.
+	// +kubebuilder:validation:XValidation:rule="has(self.name) && has(self.__namespace__)",message="both volumeGroupReplicationRef.name and volumeGroupReplicationRef.namespace must be set"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="volumeGroupReplicationRef is immutable"
+	VolumeGroupReplicationRef corev1.ObjectReference `json:"volumeGroupReplicationRef"`
+
+	// VolumeGroupReplicationHandle is a unique id returned by the CSI driver
+	// to identify the VolumeGroupReplication on the storage system.
+	VolumeGroupReplicationHandle string `json:"volumeGroupReplicationHandle"`
+
+	// provisioner is the name of the CSI driver used to create the physical
+	// volume group on
+	// the underlying storage system.
+	// This MUST be the same as the name returned by the CSI GetPluginName() call for
+	// that driver.
+	// Required.
+	Provisioner string `json:"provisioner"`
+
+	// VolumeGroupReplicationClassName is the name of the VolumeGroupReplicationClass from
+	// which this group replication was (or will be) created.
+	// +optional
+	VolumeGroupSnapshotClassName string `json:"volumeGroupReplicationClassName"`
+
+	// Source specifies whether the snapshot is (or should be) dynamically provisioned
+	// or already exists, and just requires a Kubernetes object representation.
+	// This field is immutable after creation.
+	// Required.
+	Source VolumeGroupReplicationContentSource `json:"source"`
+}
+
+// VolumeGroupSnapshotContentSource represents the CSI source of a group replication.
+type VolumeGroupReplicationContentSource struct {
+	// VolumeHandles is a list of volume handles on the backend to be grouped
+	// and replicated.
+	VolumeHandles []string `json:"volumeHandles"`
+}
+
+// VolumeGroupReplicationContentStatus defines the status of VolumeGroupReplicationContent
+type VolumeGroupReplicationContentStatus struct {
+	// PVVolumeList is the list of of PV for the group replication
+	// The maximum number of allowed PV in the group is 100.
+	// +optional
+	PVVolumeList []PVVolume `json:"pvVolumeList,omitempty"`
```

Comment:
PVVolume is a weird name, this should probably just be PersistentVolumeRefList.

PVVolume looks similar to pvVolumeSnapshotContentList, but that combines PV+VolumeSnapshotContent in each item in the list. PVVolume only contains a single Ref object to a PV.

Reply:
@Madhu-1 this is still TODO?